### PR TITLE
Enable Dynamic agent

### PR DESCRIPTION
**Pull request by sophia-guo**
Signed-off-by: Sophia Guo <sophia.gwf@gmail.com>

### CODE REVIEW

File: buildenv/jenkins/JenkinsfileBase [hunk 1]
```diff
@@ -875,9 +876,8 @@ def run_parallel_tests() {
 	if (params.PARALLEL && params.PARALLEL != "None") {
 		stage ("Parallel Tests") {
 			def childJobs = parallel parallel_tests
-
 			node {
-				def buildResult = ""
+				def buildResult = ""
```
Comment (karianna) [inline buildenv/jenkins/JenkinsfileBase:880]: whitespace

File: buildenv/jenkins/openjdk_tests [hunk 2]
```diff
@@ -144,17 +151,18 @@ def PLATFORM_MAP = [
     'x86-64_windows_mixed' : [
         'SPEC' : 'win_x86-64_mxdptrs',
         'LABEL' : 'ci.role.test&&hw.arch.x86&&sw.os.windows',
+        'DynamicAgents' : ['fyre']
     ],
     'x86-64_windows_xl' : [
         'SPEC' : 'win_x86-64',
         'LABEL' : 'ci.role.test&&hw.arch.x86&&sw.os.windows',
+        'DynamicAgents' : ['fyre']
```
Comment (llxia) [inline buildenv/jenkins/openjdk_tests:159]: We cannot start Fyre vm agents on windows atm.

**Review by smlambert — COMMENTED** [review]
LGTM, with the update to remove the windows fyre option.

File: buildenv/jenkins/openjdk_tests (hunk 2 above)
Comment (sophia-guo) [reply buildenv/jenkins/openjdk_tests]: Removed. Thanks.

**Review by smlambert — APPROVED** [review]

**Comment by sophia-guo** [comment]
@smlambert updated. We will also need to update azure VM test agent's label be `ci.role.test&&hw.arch.x86&&sw.os.linux&&ci.agent.dynamic'.

**Comment by smlambert** [comment]
re: https://github.com/AdoptOpenJDK/openjdk-tests/pull/2397#issuecomment-805877451 azuretestagent is now updated with `ci.role.test&&hw.arch.x86&&sw.os.linux&&ci.agent.dynamic` label

**Comment by smlambert** [comment]
I will also note that this PR is using [centos6_build_image](https://github.com/AdoptOpenJDK/openjdk-tests/pull/2397/files#diff-ebf20f64c486ae48bd60b965c555076519f9d5623789f4574467d3051bb530f7R303) as the test container.  

@sxa has been compiling a list of tests that do not run well in containers in https://github.com/AdoptOpenJDK/openjdk-tests/issues/2138, and making updates to static container nodes he has created.  I think it may be valuable to have a test container with any/all updates that Stewart has been making in his investigation and use it in place of the centos6_build_image.

File: buildenv/jenkins/testJobTemplate [hunk 3]
```diff
@@ -246,6 +245,7 @@ ARCH_OS_LIST.each { ARCH_OS ->
 							stringParam('UPSTREAM_TEST_JOB_NAME', "", "Auto-populated. Upstream test job name. It will be used together with PARALLEL=Dynamic")
 							stringParam('UPSTREAM_TEST_JOB_NUMBER', "", "Auto-populated. Upstream test job number. It will be used together with PARALLEL=Dynamic")
 							booleanParam('LIGHT_WEIGHT_CHECKOUT', LIGHT_WEIGHT_CHECKOUT.toBoolean(), "Optional. Default is false for Grinders")
+							stringParam('CLOUD_PROVIDER', CLOUD_PROVIDER, "available cloud provider")
```
Comment (smlambert) [inline buildenv/jenkins/testJobTemplate:248]: Minor nitpick, capitalize **A**vailable to match the other param descriptors that are capitalized

Comment (llxia) [reply buildenv/jenkins/testJobTemplate:248]: Since we are updating this line, could you also provide examples?

Comment (sophia-guo) [reply buildenv/jenkins/testJobTemplate]: Updated.

**Review by llxia — APPROVED** [review]
LGTM

File: buildenv/jenkins/openjdk_tests [hunk 4]
```diff
@@ -223,51 +233,81 @@ if (JDK_VERSIONS.size() > 1 || JDK_IMPLS.size() > 1 || PLATFORMS.size() >1 || PL
...
-                    // node(LABEL) has to be used for spinning up a dynamic vm agent
+                    // If there is available node before timeout
                     node(LABEL) {
-                        echo "find the node ${env.NODE_NAME}"
+                        echo "find the node with label as ${env.NODE_NAME}"
+                    }
+                }
+            } else {
+                boolean isNodeIdle = false
+                node('master') {
```
Comment (AdamBrousseau) [inline buildenv/jenkins/openjdk_tests:247]: Eclipse OpenJ9 Jenkins does not have any executors on master so this won't work there.

**Pull request merged**